ArkEcosystem/laravel-foundation
# refactor: improve expectValidationError helper test

## Summary
https://app.clickup.com/t/1u9thq6

Test helper `expectValidationError` passes if the callback doesn't throw any validation error and this can lead to a false positive.

In my opinion this should throws an error like "No validation error thrown." if the callback doesn't have any validation error.

## How to test
On `tests/Fortify/Actions/CreateNewUserTest.php`, scroll on line 146 and replace `password_confirmation` with `$this->validPassword` (see snippet above ⬇️ ). Without this PR test passes, with this PR test fails.

```php
it('password should match the confirmation', function () {
    Config::set('fortify.models.user', \ARKEcosystem\Foundation\Fortify\Models\User::class);

    expectValidationError(fn () => (new CreateNewUser())->create([
        'name'                  => 'John Doe',
        'username'              => 'alfonsobries',
        'email'                 => 'john@doe.com',
        'password'              => $this->validPassword,
        'password_confirmation' => $this->validPassword, // <-- replace with this line
        'terms'                 => true,
    ]), 'password_confirmation', 'The password confirmation and password must match.');
});
```







## Checklist
